[WPE] Define environment variable WEBKIT_INSPECTOR_RESOURCES_PATH in scripts

siwei-li opened the pull request:
#### 10c46db4925b67ea9936248c812a7382a2f915bc
<pre>
[WPE] Define environment variable WEBKIT_INSPECTOR_RESOURCES_PATH in scripts
<a href="https://bugs.webkit.org/show_bug.cgi?id=281771">https://bugs.webkit.org/show_bug.cgi?id=281771</a>

Reviewed by NOBODY (OOPS!).

Define the environment variable `WEBKIT_INSPECTOR_RESOURCES_PATH` to specify
the file path in places that might use the inspector resources as a
packed .gresource.

* Source/WebKit/InspectorGResources.cmake:
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Tools/MiniBrowser/wpe/CMakeLists.txt:
* Tools/MiniBrowser/wpe/main.cpp:
(main):
* Tools/MiniBrowser/wpe/qt5/main.cpp:
(main):
* Tools/MiniBrowser/wpe/qt6/main.cpp:
(main):
* Tools/Scripts/run-mvt-tests:
(WPEWebDriver.__init__):
(WPEWebDriver.start):
(WPEWebDriver._run):
* Tools/Scripts/webkit-flatpak-run-nightly:
(run):
* Tools/Scripts/webkitpy/port/glib.py:
(GLibPort.setup_environ_for_minibrowser):
* Tools/TestWebKitAPI/Tests/WPEQt/WPEQtTest.h:
* Tools/TestWebKitAPI/glib/CMakeLists.txt:
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.cpp:
(main):
</pre>

philn commented on Tools/MiniBrowser/wpe/CMakeLists.txt:
```diff
@@ -49,6 +49,8 @@ endif ()
 
 if (DEVELOPER_MODE)
     list(APPEND MiniBrowser_PRIVATE_DEFINITIONS WEBKIT_INJECTED_BUNDLE_PATH="${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
+    file(COPY "${CMAKE_BINARY_DIR}/WebInspectorUI/DerivedSources/inspector.gresource" DESTINATION "${CMAKE_BINARY_DIR}/share")
```
I suspect this should be placed elsewhere. I'll let Adrian chime in :)

aperezdc replied:
Indeed, this would belong in `Source/WebKit/PlatformWPE.cmake`. But I would rather avoid copying the file.

aperezdc reviewed (changes requested):
I think that instead of copying the `inspector.gresource` file, what is better to do is setting its output path and have the file generated directly in `${CMAKE_BINARY_DIR}/share`. Something like this (untested):

```diff
diff --git a/Source/WebKit/InspectorGResources.cmake b/Source/WebKit/InspectorGResources.cmake
index a1e3b60f8e74..5578f5448328 100644
--- a/Source/WebKit/InspectorGResources.cmake
+++ b/Source/WebKit/InspectorGResources.cmake
@@ -9,7 +9,7 @@ macro(WEBKIT_BUILD_INSPECTOR_GRESOURCES _derived_sources_dir _output_file)
     )
 
     GLIB_COMPILE_RESOURCES(
-        OUTPUT        ${_derived_sources_dir}/${_output_file}
+        OUTPUT        ${_output_file}
         SOURCE_XML    ${_derived_sources_dir}/InspectorGResourceBundle.xml
         RESOURCE_DIRS ${_derived_sources_dir}/InspectorResources/WebInspectorUI
     )
diff --git a/Source/WebKit/PlatformGTK.cmake b/Source/WebKit/PlatformGTK.cmake
index 69a8b9419318..3d41efb189b4 100644
--- a/Source/WebKit/PlatformGTK.cmake
+++ b/Source/WebKit/PlatformGTK.cmake
@@ -441,7 +441,10 @@ add_custom_command(
     VERBATIM
 )
 
-WEBKIT_BUILD_INSPECTOR_GRESOURCES(${WebKitGTK_DERIVED_SOURCES_DIR} "InspectorGResourceBundle.c")
+WEBKIT_BUILD_INSPECTOR_GRESOURCES(
+    "${WebKitGTK_DERIVED_SOURCES_DIR}"
+    "${WebKitGTK_DERIVED_SOURCES_DIR}/InspectorGResourceBundle.c"
+)
 
 set(WebKitResources "")
 list(APPEND WebKitResources "<file alias=\"css/gtk-theme.css\">gtk-theme.css</file>\n")
diff --git a/Source/WebKit/PlatformWPE.cmake b/Source/WebKit/PlatformWPE.cmake
index e2f16a9793aa..a58471aa4f4d 100644
--- a/Source/WebKit/PlatformWPE.cmake
+++ b/Source/WebKit/PlatformWPE.cmake
@@ -533,7 +533,10 @@ if (ENABLE_BREAKPAD)
     )
 endif ()
 
-WEBKIT_BUILD_INSPECTOR_GRESOURCES(${WebInspectorUI_DERIVED_SOURCES_DIR} "inspector.gresource")
+WEBKIT_BUILD_INSPECTOR_GRESOURCES(
+    "${WebInspectorUI_DERIVED_SOURCES_DIR}"
+    "${CMAKE_BINARY_DIR}/share/inspector.gresource"
+)
 
 install(FILES ${WebInspectorUI_DERIVED_SOURCES_DIR}/inspector.gresource DESTINATION "${CMAKE_INSTALL_FULL_DATADIR}/wpe-webkit-${WPE_API_VERSION}")
 
```

aperezdc commented on Source/WebKit/InspectorGResources.cmake:
```diff
@@ -9,13 +9,8 @@ macro(WEBKIT_BUILD_INSPECTOR_GRESOURCES _derived_sources_dir _output_file)
-    add_custom_target(${_output_file}
-        ALL
-        DEPENDS ${_derived_sources_dir}/${_output_file}
-    )
```
Is this really not needed now? We added it in 1fe32d327a6cccf5436deca62af9ccc7d5c85307 to ensure that the file gets always created as part of the build process.

siwei-li replied:
I thought that in line 12 the output of GLIB_COMPILE_RESOURCES is changed to the path of ` ${_output_file}` from `{_derived_sources_dir}/${_output_file}`, is it still needed?

aperezdc reviewed (commented):
Patch LGTM, with one question.

aperezdc commented on Source/WebKit/InspectorGResources.cmake:
```diff
@@ -9,13 +9,13 @@ macro(WEBKIT_BUILD_INSPECTOR_GRESOURCES _derived_sources_dir _output_file)
     )
 
-    add_custom_target(${_output_file}
+    add_custom_target(InspectorResources
```
For future reference: Making the name fixed here prevents using the macro more than once for each port; but this is fine because that's exactly how it's used at the moment.

siwei-li committed:
[WPE] Define environment variable WEBKIT_INSPECTOR_RESOURCES_PATH in scripts
https://bugs.webkit.org/show_bug.cgi?id=281771

Reviewed by Adrian Perez de Castro.

Define the environment variable `WEBKIT_INSPECTOR_RESOURCES_PATH` to specify
the file path in places that might use the inspector resources as a
packed .gresource.

* Source/WebKit/InspectorGResources.cmake:
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Tools/MiniBrowser/wpe/CMakeLists.txt:
* Tools/MiniBrowser/wpe/main.cpp:
(main):
* Tools/MiniBrowser/wpe/qt5/main.cpp:
(main):
* Tools/MiniBrowser/wpe/qt6/main.cpp:
(main):
* Tools/Scripts/generate-bundle:
(BundleCreator._generate_wrapper):
* Tools/Scripts/run-mvt-tests:
(WPEWebDriver.__init__):
(WPEWebDriver.start):
(WPEWebDriver._run):
* Tools/Scripts/webkit-flatpak-run-nightly:
(run):
* Tools/Scripts/webkitpy/port/glib.py:
(GLibPort.setup_environ_for_minibrowser):
* Tools/TestWebKitAPI/Tests/WPEQt/WPEQtTest.h:
* Tools/TestWebKitAPI/glib/CMakeLists.txt:
* Tools/TestWebKitAPI/glib/WebKitGLib/TestMain.cpp:
(main):

Canonical link: https://commits.webkit.org/286326@main
